File: leetcode14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        n = len(strs)
        if n == 0 or len(strs[0]) == 0: return ""
        for j in range(1, len(strs[0]) + 2):
            for i in range(1, n):
                if len(strs[i]) < j:
                    return strs[0][:j - 1]
                if strs[i][:j] != strs[0][:j]:
                    return strs[0][:j - 1]
        return strs[0][:j - 1]

Runtime: 20 ms, faster than 93.94% of Python online submissions for Longest Common Prefix.
找到最短的str
shortest = min(strs, key=len)  “flow”

File: leetcode14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n=strs.size(),i,j;
        if(n==0 || strs[0].length()==0) return "";
        string res="";
        for (i=0;i<strs[0].length();i++){
            for (j=0;j<n;j++){
                if(strs[j].length()<i) return res;
                if(strs[j][i]!=strs[0][i]) return res;
            }
            res+=strs[0][i];
        }
        return res;
    }
};
两种情况：1.长度不够了 "fl" "flow"
2.字符不同
Runtime: 8 ms, faster than 94.25% of C++ online submissions for Longest Common Prefix.